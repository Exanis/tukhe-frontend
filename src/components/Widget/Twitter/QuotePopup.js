import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { injectIntl, defineMessages } from 'react-intl';
import TweetInputBox from './TweetInputBox';
import Tweet from './Tweet';

const messages = defineMessages({
    quoteMessageLabel: {
        id: 'widget.twitter.tweet.quoteMessageLabel',
        defaultMessage: 'Add a message'
    },
});

export class RawQuotePopup extends Component {
    render () {
        return <div>
            <TweetInputBox
                onUpdateText={this.props.onUpdateText}
                label={this.props.intl.formatMessage(messages.quoteMessageLabel)}
                />
            <Tweet
                tweet={this.props.tweet}
                retweeted={this.props.tweet.author.screen_name}
                isQuote={false}
                />
        </div>;
    }
}

const QuotePopup = injectIntl(RawQuotePopup);

QuotePopup.propTypes = {
    onUpdateText: PropTypes.func.isRequired,
    tweet: PropTypes.shape({
        author: PropTypes.shape({
            screen_name: PropTypes.string
        })
    })
};

RawQuotePopup.propTypes = {
    ...QuotePopup.propTypes,
    intl: PropTypes.object.isRequired
};

export default QuotePopup;
