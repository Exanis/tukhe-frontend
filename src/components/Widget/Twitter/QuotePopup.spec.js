import React from 'react';
import { shallow } from 'enzyme';
import { RawQuotePopup } from "./QuotePopup";

const props = {
    onUpdateText: jest.fn(),
    tweet: {
        author: {
            screen_name: 'test'
        }
    },
    intl: {
        formatMessage: () => 'text'
    }
};

it('Should render correctly', () => {
    const wrapper = shallow(<RawQuotePopup {...props} />);

    expect(wrapper).toMatchSnapshot();
});
