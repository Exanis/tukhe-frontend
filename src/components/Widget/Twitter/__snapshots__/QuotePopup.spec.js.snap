// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RawQuotePopup
    intl={
      Object {
        "formatMessage": [Function],
      }
    }
    onUpdateText={[MockFunction]}
    tweet={
      Object {
        "author": Object {
          "screen_name": "test",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TweetInputBox
          label="text"
          onUpdateText={[MockFunction]}
        />,
        <WithStyles(InjectIntl(RawTweet))
          isQuote={false}
          retweeted="test"
          tweet={
            Object {
              "author": Object {
                "screen_name": "test",
              },
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "label": "text",
          "onUpdateText": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "isQuote": false,
          "retweeted": "test",
          "tweet": Object {
            "author": Object {
              "screen_name": "test",
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TweetInputBox
            label="text"
            onUpdateText={[MockFunction]}
          />,
          <WithStyles(InjectIntl(RawTweet))
            isQuote={false}
            retweeted="test"
            tweet={
              Object {
                "author": Object {
                  "screen_name": "test",
                },
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "label": "text",
            "onUpdateText": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isQuote": false,
            "retweeted": "test",
            "tweet": Object {
              "author": Object {
                "screen_name": "test",
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
